js/authentication.md

```diff
@@ -1662,6 +1550,140 @@ Note: To work with Service Interface Objects, your Amazon Cognito users' [IAM ro
+
+#### Migrating Users to AWS Cognito
+
+One of the reason to use this authentication flow is that it would allow you to migrate users to AWS Cognito.
+
+Amazon Cognito provides a trigger to migrate users from your existing user directory to Cognito seamlessly. You configure your Cognito User Pool's "Migration" trigger to invoke a Lambda function whenever a user that does not already exist in the user pool signs in or resets their password. 
+
+In short, the Lambda function should validate the user credentials against your existing user directory and return a response object containing user attributes and status on success, or an error message on error. There's a good documentation [here](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-import-using-lambda.html) on how to set up this migration flow and a more detailed instruction [here](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-migrate-user.html#cognito-user-pools-lambda-trigger-syntax-user-migration) on how the lambda should handle request and response objects.
+
+### Customizing Authentication Flow
+
+Amazon Cognito User Pools support customizing the authentication flow to enable new challenge types, in addition to a password, to verify the identity of users. These challenge types may include CAPTCHAs or dynamic challenge questions.
+
+To define your challenges for custom authentication flow, you need to implement Lambda triggers for Amazon Cognito.
+
+For more information about working with Lambda Triggers for custom authentication challenge, please visit [Amazon Cognito Developer Documentation](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-challenge.html).
+{: .callout .callout--info}
+
+#### How to use it in Amplify
+
+To initiate a custom authorization flow in your app, call `signIn` without a password. A custom challenge needs to be answered using the `sendCustomChallengeAnswer` method:
+
+```javascript
+import { Auth } from 'aws-amplify';
+
+Auth.configure({
+    // other configurations
+    // ...
+    authenticationFlowType: 'CUSTOM_AUTH'
+});
+
+let challengeResponse = "the answer for the challenge";
+
+Auth.signIn(username, password)
+    .then(user => {
+        if (user.challengeName === 'CUSTOM_CHALLENGE') {
+            // to send the answer of the custom challenge
+            Auth.sendCustomChallengeAnswer(user, challengeResponse)
+                .then(user => console.log(user))
+                .catch(err => console.log(err));
+        } else {
+            console.log(user);
+        }
+    })
+    .catch(err => console.log(err));
+```
+
+#### Creating a CAPTCHA
+
+Here is the sample of how to create a CAPTCHA challenge with the Lambda Triggers.
```

Comment:
Here is a sample for creating a CAPTCHA challenge with a Lambda Trigger

Reply:
Done.